### PR TITLE
Add vertCruiseSpeed to craft type

### PR DESCRIPTION
**Pull request by TylerS1066**
Adds a vertCruiseSpeed to the craft type config.

### Checklist
- [ ] Unit tests 
- [x] Proper internationalization
- [x] Compiled/tested

### CODE REVIEW

**Comment by cccm5** [comment]
So just to make sure of what I'm reading, this basically ensures that if a craft moves both vertically and horizontally it won't move at a higher total speed than if it was doing either individually, correct?

**Comment by TylerS1066** [comment]
> So just to make sure of what I'm reading, this basically ensures that if a craft moves both vertically and horizontally it won't move at a higher total speed than if it was doing either individually, correct?

No, that has already been implemented in an earlier commit.  This changes the vertical cruise speed from being cruiseSpeed to being cruiseSpeed scaled by the skip block ratio.

As an example, Airships move 4 blocks horizontally at 4 m/s and 2 blocks vertically, so they will move vertically at 2 m/s.

**Comment by TylerS1066** [comment]
After further review, I decided to add a vertCruiseSpeed configuration to the craft type, allowing individual tweaking of the vertical speed independent of the skip block ratios of the craft.

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java [hunk 1]
```diff
@@ -421,14 +421,16 @@ private void processCruise() {
             if(Settings.Debug) {
                 Movecraft.getInstance().getLogger().info("TickCoolDown: " + tickCoolDown);
             }
-            if(bankLeft || bankRight) {
-                if (!dive) {
-                    tickCoolDown *= (Math.sqrt(Math.pow(1 + pcraft.getType().getCruiseSkipBlocks(), 2) + Math.pow(pcraft.getType().getCruiseSkipBlocks() >> 1, 2)) / (1 + pcraft.getType().getCruiseSkipBlocks()));
-                } else {
-                    tickCoolDown *= (Math.sqrt(Math.pow(1 + pcraft.getType().getCruiseSkipBlocks(), 2) + Math.pow(pcraft.getType().getCruiseSkipBlocks() >> 1, 2) + 1) / (1 + pcraft.getType().getCruiseSkipBlocks()));
+            if(pcraft.getCruiseDirection() != 0x42 && pcraft.getCruiseDirection() != 0x43) {
```
Comment (TylerS1066) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java:424]: This change is a bugfix to a previous bug that changed the tickCooldown of an ascending or descending craft if the player was banking or diving.

File: modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java [hunk 2]
```diff
@@ -376,8 +373,13 @@ public int getTickCooldown() {
      * gets the speed of a craft in blocks per second.
      * @return the speed of the craft
      */
-    public double getSpeed(){
-        return 20*(type.getCruiseSkipBlocks()+1)/(double)getTickCooldown();
+    public double getSpeed() {
+        if(cruiseDirection == 0x42 || cruiseDirection == 0x43) {
```
Comment (TylerS1066) [inline modules/api/src/main/java/net/countercraft/movecraft/craft/Craft.java:377]: This is needed to correctly adjust the speed display based on which skipBlocks is being used.

File: modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java [hunk 3]
```diff
@@ -161,10 +162,14 @@ public CraftType(File f) {
 
         double cruiseSpeed = doubleFromObject(data.getOrDefault("cruiseSpeed", 20.0 / tickCooldown));
         cruiseTickCooldown = (int) Math.round((1.0 + cruiseSkipBlocks) * 20.0 / cruiseSpeed);
+        double vertCruiseSpeed = doubleFromObject(data.getOrDefault("vertCruiseSpeed", 20.0 / cruiseTickCooldown));
```
Comment (cccm5) [inline modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java:165]: use cruiseSpeed as the default

Comment (TylerS1066) [reply modules/api/src/main/java/net/countercraft/movecraft/craft/CraftType.java]: Requested change made

**Pull request merged**